go/vt/vtgate/executor.go

```diff
@@ -231,7 +235,12 @@ func (e *Executor) Execute(ctx context.Context, mysqlCtx vtgateservice.MySQLConn
 		if err != nil {
 			piiSafeSQL = logStats.StmtType
 		}
-		log.Warningf("%q exceeds warning threshold of max memory rows: %v. Actual memory rows: %v", piiSafeSQL, warnMemoryRows, len(result.Rows))
+		warningMsg := fmt.Sprintf("%q exceeds warning threshold of max memory rows: %v. Actual memory rows: %v", piiSafeSQL, warnMemoryRows, len(result.Rows))
+		exceedMemoryRowsLogger.Warningf(sql, warningMsg)
+		safeSession.RecordWarning(&querypb.QueryWarning{
+			Code:    uint32(sqlerror.EROutOfMemory),
```

Comment:
Not sure if `EROutOfMemory` is appropriate, we can also remove the `Code` altogether